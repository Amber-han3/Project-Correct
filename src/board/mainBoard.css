.boardPic{
    background-repeat:no-repeat;
    /* height: 100vh;
    width: 100vw; */

    /* 背景圖片等比例縮放 */
    background-size: cover;
    background-position:center;
    /* position: relative; */
    /* background-attachment: scroll; */
    background-attachment: fixed;
    min-height: 100vh;
    height: 100%;

}

.postboardAll{
    max-width:1100px;
    /* margin-top:100px;
    margin-left:50px; */

    padding-top: 100px;
    padding-left: 50px;
    padding-bottom: 100px;
    padding-right: 20px;
    /* overflow: hidden; */

    display: flex;
    justify-content: center;
}

.postList{
    width: 50%;
    margin-right: 20px;
    justify-content: center;
    max-width: 500px;
}

.postarea{
    /* width: 50%; */
    position: relative;
    overflow: hidden;
}

/* #postDiv { */
textarea {
    width: 100%;
    height: 150px;
    font-size: 18px;
    font-family: 微軟正黑體, Microsoft JhengHei;
    border-radius: 10px;
    opacity: 0.95;
}

.postBtn{
    height: 30px;
    border-width: 1px;
    border-style: solid;
    border-radius: 5px;
    display: inline-block;
    /* position:absolute; */
    right: 0;

    background-color: white;
}

.textBoxList{
    /* width: 50%; */
    margin-top: 50px;
}


.textBoxNormal{
    /* height: auto; */
    min-height: 70px;
    border-width: 1px;
    border-color: darkgrey;
    border-style: solid;
    border-radius: 5px;
    margin: 10px 0px;
    /* padding-left: 10px;
    padding-top: 10px; */
    padding: 10px 10px;
    box-shadow:3px 3px 9px lightgrey;

    background-color: white;
    opacity: 0.8;
    
    /* 文字換行 */
    word-wrap:break-word; 
}

.textBoxNormal:hover{
    opacity: 1;
}

.textBoxEdit{
    height: auto;
    width: 100%;
    border-style: solid;
    border-width: 1px;
    border-radius:5px;
    border-color: darkgrey;
    background-color: #e9e9e9;
    color: red;
    margin-bottom: 10px;
    /* padding-left: 10px;
    padding-top: 10px; */
    padding: 10px 10px;

    /* 文字換行 */
    word-wrap:break-word; 
}

.editBoxShow{
    display: flex;
    height: 30px;
    width:100%;
    justify-content: flex-end;
    /* margin-bottom: 20px; */
    margin-bottom: 5px;
}

.editBoxHide{
    display: none;
    height: 30px;
    width:100%;
    justify-content: flex-end;
    margin-bottom: 20px;
}

.editBoxItem{
    display: inline-block;
    letter-spacing: 1px;
    margin: 0px 5px;
    border-style: solid;
    border-width: 1px;
    border-color: darkgrey;
    border-radius: 5px;
    padding: 3px;
    height: 24px;
    width: 40px;
}

.editBoxItem:hover{
    cursor: pointer;
}

/* 回覆欄位 */
.replyList{
    display: flex;
    width: 30%;
    justify-content: flex-end;
    margin-left: 20px;
}

.replyInputDiv{

    width: 100%;
    min-height: 250px;
    height: 250px;
    font-size: 18px;
    font-family: 微軟正黑體, Microsoft JhengHei;
    border-radius: 5px;
    opacity: 0.9;

    background-color: white;
    padding: 15px;
}