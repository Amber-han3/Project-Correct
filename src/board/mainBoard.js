import React, { Component } from "react";
// import ReactDOM from "react-dom";
// import {BrowserRouter as Router, Route, Link} from "react-router-dom";
import "./mainBoard.css";
import BoardList from "./boardList";

import firebase from './firebase';

// firebase相關

const db = firebase.firestore();

class MainBoard extends Component{
    constructor(props){
        super(props);
        this.state={text:"", commentWithID:[],  //紀錄文字內容用
        revise:"false",
        };  

        // 讀取firebase資料
        db.collection("users").get().then((querySnapshot) => {
            querySnapshot.forEach((doc) => {
                // 可以跨doc讀到全部comment欄位
                console.log(`${doc.id} => ${doc.data().comment}`);

                // 把讀取到的id存到state
                this.state.commentWithID.push({commentID: doc.id, comment: doc.data().comment});
                this.setState({commentWithID:this.state.commentWithID});
                console.log("commentWithID:"+this.state.commentWithID);
                // console.log("commentID:"+this.state.commentWithID[0].commentID);
                // console.log("comment:"+this.state.commentWithID[0].comment);
            });
        });
        
    }

    handleTextChange(e){
        // this.setState({text: this.state.text});
    }

    addNewPost(e){
        e.preventDefault();
        let postText=document.getElementById("textareaDiv").value
        // this.setState({text: postText});
        // console.log(this.state.text);

        // firebase相關

        const firebase = require("firebase");
        // Required for side-effects
        require("firebase/firestore");

        // 把回覆存進database

        const submitTime = new Date()

        db.collection("users").add({
            time: submitTime,
            userID: "no",
            comment: postText,
        })
        .then((docRef) => {
             console.log("Document written with comment: ", docRef.id.comment);

            // 取出db內容傳入新陣列，再放入state(避免重複render)

             db.collection("users").get().then((querySnapshot) => {
                let data=[];
                querySnapshot.forEach((doc) => {
    
                    //this.state.commentWithID.push({commentID: doc.id, comment: doc.data().comment});
                    //this.setState({commentWithID:this.state.commentWithID});
    
                    data.push({commentID: doc.id, comment: doc.data().comment});
                    console.log(data);
                });
                console.log(this.state.commentWithID);
                this.setState({commentWithID:data});
            });         

        })
        .catch(function(error) {
            console.error("Error adding document: ", error);
        });

    } 

    remove(e){
		e.preventDefault();
        
        // 在state裡的id
        let indexValue=event.target.getAttribute("id");
        let textList= [...this.state.commentWithID]
        textList.splice(indexValue, 1); 
        this.setState({commentWithID:textList});

        // firbase的處理
        let itemID=event.target.getAttribute("commentID");

        // 同步刪除firebase內容
        db.collection("users").doc(itemID).delete();

	}

	revise(e){
		e.preventDefault();
        // alert("revise");
        
        let itemID=event.target.getAttribute("commentID");
        alert(itemID);

        // this.setState({revise: "true"});
    }

    saveChange(e){
        this.setState({revise: "false"});
    }

    render(){

        let newPostDiv = this.state.commentWithID.map((text, index)=>{
                return  <div id={index} className="textBox">
                            <div id={index} className="textBoxNormal" 
                            commentID={text.commentID} >{text.comment}

                                <div id={index} className="editBoxShow">
                                    <div id={index} commentID={text.commentID} 
                                    className="editBoxItem" onClick={this.revise.bind(this)}>訂正</div>
                                    <div id={index} commentID={text.commentID} 
                                    className="editBoxItem" onClick={this.remove.bind(this)}>刪除</div>
                                </div>
                                
                            </div>
                        </div>
        })

        return(<div className="postboardAll">
                    {/* <div className="postboardPos"> */}
                        <div className="postarea">
                            <textarea placeholder="分享點什麼吧" id="textareaDiv"
                            // value={this.state.text}
                            onChange={this.handleTextChange.bind(this)}></textarea>
                            <button className="postBtn" onClick={this.addNewPost.bind(this)}>送出</button>
                        </div>
                        <div className="textBoxList">
                            <div>新增欄位從這開始</div>
                            {newPostDiv}
                        </div>
                        {/* <div>總樣式種類預覽</div>
                        <BoardList /> */}
                    {/* </div> */}
                </div>
        )
        
    }
}

export default MainBoard;
