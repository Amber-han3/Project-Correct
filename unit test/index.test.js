import React from 'react';
import renderer from 'react-test-renderer';

import mainBoard from "../src/board/mainBoard";

// 確認mainBoard UI顯示正常

it('mainBoard render correctly', () => {
    const showBoard = renderer
      .create(<mainBoard></mainBoard>)
      .toJSON()
    expect(showBoard).toMatchSnapshot()
})
