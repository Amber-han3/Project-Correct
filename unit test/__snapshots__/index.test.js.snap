// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mainBoard render correctly 1`] = `<mainBoard />`;
